src/issue/issue.ts

```diff
@@ -46,8 +49,29 @@ async function internal_issueVerifiableCredential(
   issuerEndpoint: Iri,
   subjectClaims: JsonLd,
   credentialClaims?: JsonLd,
-  options?: OptionsType,
-): Promise<VerifiableCredential> {
+  options?: {
+    fetch?: typeof fallbackFetch;
+    returnLegacyJsonld?: true;
+  } & ParseOptions,
+): Promise<VerifiableCredential>;
+async function internal_issueVerifiableCredential(
+  issuerEndpoint: Iri,
+  subjectClaims: JsonLd,
+  credentialClaims?: JsonLd,
+  options?: {
+    fetch?: typeof fallbackFetch;
+    returnLegacyJsonld?: boolean;
```

Comment:
If we don't have a type overload for `boolean` and only one type overload for `true` and one type overload for `false` then we end up with a type error when the value passed in is of type `boolean` (in this case the input of `issueVerifiableCredential`  which calls `internal_issueVerifiableCredential` will be `boolean` in the input signature rather than explicitly true or false.).

Instead we just put the `?: true` case first and mark that as deprecated and then everything else is `?: boolean` because the type overloads are evaluated in order of appearance so.

Note this ordered evaluated of type also makes https://github.com/inrupt/solid-client-vc-js/pull/849#discussion_r1405936803 a little harder because of the fact that we want to catch the `?: true` case and then catch everything else with `?: boolean`.

My proposal to resolve this and https://github.com/inrupt/solid-client-vc-js/pull/849#discussion_r1405936803 is that we have 3 overloads in the following order.

`: false`: Not deprecated and used for the JSDoc
`?: true`: Marked as deprecated
`?: boolean`: Used to catch when cases there is a boolean parameter that is not yet known to be `true` or `false`. Possibly mark as deprecated as it does have the possibility of being `true`.

Reply:
Sounds good.